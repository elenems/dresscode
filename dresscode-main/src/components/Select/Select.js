import ReactSelect from "react-select";
import React from "react";
import PropTypes from "prop-types";

function Select(props) {
  return (
    <div className="select">
      <span className="label-style">{props.label}</span>
      <ReactSelect
        value={props.value}
        onChange={props.handleChange}
        options={props.options}
      />
    </div>
  );
}

Select.propTypes = {
  handleChange: PropTypes.func.isRequired,
  options: PropTypes.array,
  label: PropTypes.string
};

export default React.memo(Select);
