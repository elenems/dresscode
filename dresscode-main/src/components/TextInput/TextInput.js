import React from "react";
import PropTypes from "prop-types";
function TextInput(props) {
  const { name, place, value, handleChange, type, error, required } = props;
  const inputStyle = error
    ? { borderColor: "#d0021b" }
    : { borderColor: "#b1b1b1" };
  return (
    <div className="input-container">
      {required === false ? (
        <label className="required" htmlFor={name}>
          {name}
        </label>
      ) : (
        <label htmlFor={name}>{name}</label>
      )}
      <input
        style={inputStyle}
        onChange={handleChange}
        id={name}
        type={type}
        value={value}
        placeholder={place}
      />
      {error ? <span className="error">{error}</span> : null}
    </div>
  );
}

TextInput.propTypes = {
  name: PropTypes.string.isRequired,
  place: PropTypes.string.isRequired,
  value: PropTypes.string,
  text: PropTypes.string,
  error: PropTypes.string,
  handleChange: PropTypes.func.isRequired
};

export default React.memo(TextInput);
