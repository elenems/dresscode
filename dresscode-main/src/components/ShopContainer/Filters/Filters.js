import React from "react";
import TextInput from "../../TextInput/TextInput";
import Select from "../../Select/Select";
import PropTypes from "prop-types";

const conditionOptions = [
  { value: "New", label: "New" },
  { value: "Perfect", label: "Perfect" },
  { value: "Good", label: "Good" },
  { value: "Average", label: "Average" },
  { value: "All", label: "All" }
];
function Filters(props) {
  return (
    <div className="filters">
      <div className="filter">
        <div className="input-block">
          <TextInput
            handleChange={props.handleChange}
            type="text"
            place="Clothes brand"
            value={props.filters.Brand}
            name="Brand"
          />
        </div>
        <div className="input-block">
          <Select
            label="Choose a condition"
            value={props.condition}
            handleChange={props.handleConditionChange}
            options={conditionOptions}
          />
        </div>
        <div className="input-block">
          <div className="multiple-filters-in-row">
            <TextInput
              handleChange={props.handleChange}
              type="text"
              place="$0"
              value={props.filters.PriceFrom}
              name="From"
            />
            <TextInput
              handleChange={props.handleChange}
              type="text"
              place="$500"
              value={props.filters.PriceTo}
              name="To"
            />
          </div>
        </div>
        <div className="input-block">
          <TextInput
            handleChange={props.handleChange}
            type="text"
            place="Clothes size"
            value={props.filters.Size}
            name="Size"
          />
        </div>
      </div>
    </div>
  );
}

Filters.propTypes = {
  handleChange: PropTypes.func.isRequired,
  filters: PropTypes.object.isRequired,
  handleConditionChange: PropTypes.func.isRequired,
  condition: PropTypes.object
};

export default React.memo(Filters);
