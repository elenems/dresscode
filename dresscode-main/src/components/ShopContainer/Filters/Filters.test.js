import { shallow } from "enzyme";
import React from "react";
import Filters from "./Filters";
import TextInput from "../../TextInput/TextInput";
import Select from "../../Select/Select";

describe("Filters", () => {
  const props = {
    filters: {
      Brand: "Adidas"
    }
  };
  it("Matches snapshot", () => {
    const wrapper = shallow(<Filters {...props} />).debug();
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders 4 TextInputs", () => {
    const wrapper = shallow(<Filters {...props} />);
    expect(wrapper.find(TextInput).length).toBe(4);
  });

  it("Renders 1 Select", () => {
    const wrapper = shallow(<Filters {...props} />);
    expect(wrapper.find(Select).length).toBe(1);
  });
});
