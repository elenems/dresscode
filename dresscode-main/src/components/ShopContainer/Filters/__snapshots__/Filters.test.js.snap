// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters Matches snapshot 1`] = `
"<div className=\\"filters\\">
  <div className=\\"filter\\">
    <div className=\\"input-block\\">
      <Memo(TextInput) handleChange={[undefined]} type=\\"text\\" place=\\"Clothes brand\\" value=\\"Adidas\\" name=\\"Brand\\" />
    </div>
    <div className=\\"input-block\\">
      <Memo(Select) label=\\"Choose a condition\\" value={[undefined]} handleChange={[undefined]} options={{...}} />
    </div>
    <div className=\\"input-block\\">
      <div className=\\"multiple-filters-in-row\\">
        <Memo(TextInput) handleChange={[undefined]} type=\\"text\\" place=\\"$0\\" value={[undefined]} name=\\"From\\" />
        <Memo(TextInput) handleChange={[undefined]} type=\\"text\\" place=\\"$500\\" value={[undefined]} name=\\"To\\" />
      </div>
    </div>
    <div className=\\"input-block\\">
      <Memo(TextInput) handleChange={[undefined]} type=\\"text\\" place=\\"Clothes size\\" value={[undefined]} name=\\"Size\\" />
    </div>
  </div>
</div>"
`;
